src/query-builder/builder/SelectQueryBuilder.ts

```diff
@@ -1182,43 +1128,11 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         if (this.expressionMap.lockMode === "optimistic")
             throw new OptimisticLockCanNotBeUsedError();
 
-        const queryRunner = this.obtainQueryRunner();
-        let transactionStartedByUs: boolean = false;
-        try {
-
-            // start transaction if it was enabled
-            if (this.expressionMap.useTransaction === true && queryRunner.isTransactionActive === false) {
-                await queryRunner.startTransaction();
-                transactionStartedByUs = true;
-            }
-
-            this.expressionMap.queryEntity = true;
-            const entitiesAndRaw = await this.executeEntitiesAndRawResults(queryRunner);
-            this.expressionMap.queryEntity = false;
-            const count = await this.executeCountQuery(queryRunner);
-            const results: [Entity[], number] = [entitiesAndRaw.entities, count];
```

Comment:
New code could be made nicer perhaps... Counts are somewhat the exception here requiring two queries in this way.